gluten-substrait/src/main/scala/org/apache/gluten/extension/columnar/OffloadSingleNode.scala

```diff
@@ -332,6 +332,8 @@ case class OffloadProject() extends OffloadSingleNode with LogLevelUtil {
           expr => rewriteExpr(expr, replacedExprs).asInstanceOf[NamedExpression]
         }
         val newChild = addMetadataCol(projectExec.child, replacedExprs)
+        // tag child again, addMetadataCol lost the tag.
+        addHint.apply(newChild)
```

Comment:
@zhztheplayer please help review this code, fixed an issue where fallback could not be performed correctly